src/java.base/share/classes/sun/nio/ch/NioSocketImpl.java

```diff
@@ -182,6 +182,12 @@ private void park(FileDescriptor fd, int event, long nanos) throws IOException {
                 millis = -1;
             } else {
                 millis = NANOSECONDS.toMillis(nanos);
+                if (nanos > MILLISECONDS.toNanos(millis)) {
+                    // Round up any excess nanos to the nearest millisecond. Avoids
+                    // parking for less than requested, especially when poll is
+                    // called with zero millis, which would return immediately.
```

Comment:
Rounding down to 0 was an oversight so thanks for fixing this. The comment is a bit excessive, can you reduce it down to "Round up any excess nanos to the nearest millisecond to avoid parking for less than requested".